broker/geo_accession/routes.py

```diff
@@ -0,0 +1,38 @@
+# --- core imports
+from http import HTTPStatus
+import tempfile
+
+# --- application imports
+from broker.common.util import response_json
+
+# --- third-party imports
+from flask import Blueprint, send_file, request
+from flask_cors import cross_origin
+from geo_to_hca import geo_to_hca
+
+geo_accession_bp = Blueprint(
+    'geo_accession', __name__, url_prefix='/geo-accession'
+)
+
+
+@cross_origin()
+@geo_accession_bp.route("/spreadsheet")
+def get_spreadsheet_using_geo():
+    args = request.args
+    geo_accession = args.get('accession')
+
+    try:
+        workbook = geo_to_hca.create_spreadsheet_using_geo_accession(geo_accession)
+
+        if workbook:
+            temp_file = tempfile.NamedTemporaryFile()
+            filename = f"hca_metadata_spreadsheet-{geo_accession}.xlsx"
+            workbook.save(temp_file.name)
+
+            return send_file(temp_file.name,
+                             as_attachment=True,
+                             cache_timeout=0,
+                             attachment_filename=filename)
+
+    except Exception:
+        return response_json(HTTPStatus.INTERNAL_SERVER_ERROR, "Unable to find HCA metadata against given accession")
```

Comment:
@amnonkhen 

I changed this from `HTTPStatus.NOT_FOUND` to `HTTPStatus.INTERNAL_SERVER_ERROR` but the http interceptor on the UI interprets this as a service being down, and navigates to a page that looks like 

![image](https://user-images.githubusercontent.com/38984808/150761711-5c7e081c-5f13-4a06-908d-6240b71edf33.png)

I think I might have to go with `HTTPStatus.NOT_FOUND`, for now, since we currently don't have specific exceptions to catch from the geo-to-hca. The most likely scenario is that the wranglers enter an incorrect accession and it doesn't make sense to navigate to above page in that scenario

Reply:
@amnonkhen Why did you want to change 404 to 500? Returning 500 is always a bad idea, if there is something going wrong on our side then it is better to specify what went wrong. 500 is just a generic response and not helpful for anybody.
If I understand correctly, in this case 404 would be better as the no metadata can be found by the user given parameters. Am I wrong?

Reply:
as discussed with @amnonkhen , I've updated the code so that there's a check for geo accession validity. If the accession is invalid, the HTTP code 400 is returned. In case of any other exception, HTTP code 500 is returned

Reply:
@amnonkhen @yusra-haider Well, returning 400 and 500 is not the best thing to do. Why can't we specify what is the exact problem?

Reply:
We are using the HTTP status codes to indicate what the problem is. If the accession is invalid, the HTTP status code of 400 makes sense. Other than that, the geo-to-hca library is a blackbox, it doesn't throw any specific exceptions so we can't really capture the exact issue.

cc: @amnonkhen